### PR TITLE
Source Facebook Marketing: update spec descriptions

### PR DESCRIPTION
**Pull request by ChristoGrab**
## What
Updates to the Facebook Marketing spec.

## How
- Added an explicit order to the Custom Insights properties. I followed the order in which they were added in the spec file, as it felt like an intuitive progression.
- Rewrote a number of descriptions for clarity.

### CODE REVIEW

**Comment by octavia-squidington-iii** [comment]
##  source-facebook-marketing test report (commit [`cb7505b664`](https://github.com/airbytehq/airbyte/pull/28948/commits/cb7505b6642d8c450c1af8c1336d9aba117c0308)) - ❌

⏲️  Total pipeline duration: 63mn25s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-facebook-marketing/metadata.yaml | ✅        |
| Connector version semver check                                                   | ✅        |
| Connector version increment check                                                | ✅        |
| QA checks                                                                        | ✅        |
| Code format checks                                                               | ❌        |
| Connector package install                                                        | ✅        |
| Build source-facebook-marketing docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                       | ✅        |
| Integration tests                                                                | ✅        |
| Acceptance tests                                                                 | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci-internal/connectors/test/pull_request/facebook-marketing-spec/1691079245/cb7505b6642d8c450c1af8c1336d9aba117c0308/source-facebook-marketing/1.1.2/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-facebook-marketing test
```

**Comment by octavia-squidington-iii** [comment]
##  source-facebook-marketing test report (commit [`dd8db6126d`](https://github.com/airbytehq/airbyte/pull/28948/commits/dd8db6126d70a6c3bdfb7d6bf7721a41620df8f6)) - ❌

⏲️  Total pipeline duration: 67mn10s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-facebook-marketing/metadata.yaml | ✅        |
| Connector version semver check                                                   | ✅        |
| Connector version increment check                                                | ❌        |
| QA checks                                                                        | ✅        |
| Code format checks                                                               | ❌        |
| Connector package install                                                        | ✅        |
| Build source-facebook-marketing docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                       | ✅        |
| Integration tests                                                                | ✅        |
| Acceptance tests                                                                 | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/facebook-marketing-spec/1691414222/dd8db6126d70a6c3bdfb7d6bf7721a41620df8f6/source-facebook-marketing/1.1.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:dd8db6126d70a6c3bdfb7d6bf7721a41620df8f6)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-facebook-marketing test
```

**Comment by octavia-squidington-iii** [comment]
##  source-facebook-marketing test report (commit [`4bb55448eb`](https://github.com/airbytehq/airbyte/pull/28948/commits/4bb55448ebd26de7688203c9ff8571779f9bc915)) - ❌

⏲️  Total pipeline duration: 73mn32s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-facebook-marketing/metadata.yaml | ✅        |
| Connector version semver check                                                   | ✅        |
| Connector version increment check                                                | ✅        |
| QA checks                                                                        | ✅        |
| Code format checks                                                               | ❌        |
| Connector package install                                                        | ✅        |
| Build source-facebook-marketing docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                       | ✅        |
| Integration tests                                                                | ✅        |
| Acceptance tests                                                                 | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/facebook-marketing-spec/1691418385/4bb55448ebd26de7688203c9ff8571779f9bc915/source-facebook-marketing/1.1.3/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:4bb55448ebd26de7688203c9ff8571779f9bc915)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-facebook-marketing test
```

**Comment by octavia-squidington-iii** [comment]
##  source-facebook-marketing test report (commit [`99b4f29f66`](https://github.com/airbytehq/airbyte/pull/28948/commits/99b4f29f66d012e1ba911ae7d2a517035092c33d)) - ❌

⏲️  Total pipeline duration: 65mn18s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-facebook-marketing/metadata.yaml | ✅        |
| Connector version semver check                                                   | ✅        |
| Connector version increment check                                                | ✅        |
| QA checks                                                                        | ✅        |
| Code format checks                                                               | ❌        |
| Connector package install                                                        | ✅        |
| Build source-facebook-marketing docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                       | ✅        |
| Integration tests                                                                | ✅        |
| Acceptance tests                                                                 | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/facebook-marketing-spec/1691508032/99b4f29f66d012e1ba911ae7d2a517035092c33d/source-facebook-marketing/1.1.3/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:99b4f29f66d012e1ba911ae7d2a517035092c33d)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-facebook-marketing test
```

**Comment by sherifnada** [comment]
@ChristoGrab could you fix the merge conflict? I updated the PR so it _should_ fix the breaking build

**Review by sherifnada — APPROVED** [review]

**Comment by octavia-squidington-iii** [comment]
##  source-facebook-marketing test report (commit [`068ca0f552`](https://github.com/airbytehq/airbyte/pull/28948/commits/068ca0f55264f230e2a7612960f4154b27fa68c3)) - ❌

⏲️  Total pipeline duration: 53mn42s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-facebook-marketing/metadata.yaml | ✅        |
| Connector version semver check                                                   | ✅        |
| Connector version increment check                                                | ✅        |
| QA checks                                                                        | ✅        |
| Code format checks                                                               | ✅        |
| Connector package install                                                        | ✅        |
| Build source-facebook-marketing docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                       | ✅        |
| Integration tests                                                                | ✅        |
| Acceptance tests                                                                 | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/facebook-marketing-spec/1691790265/068ca0f55264f230e2a7612960f4154b27fa68c3/source-facebook-marketing/1.1.3/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:068ca0f55264f230e2a7612960f4154b27fa68c3)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-facebook-marketing test
```

**Comment by octavia-squidington-iii** [comment]
##  source-facebook-marketing test report (commit [`3a4ce7179d`](https://github.com/airbytehq/airbyte/pull/28948/commits/3a4ce7179ded844658c07e09fa6cd51df97652a3)) - ❌

⏲️  Total pipeline duration: 63mn34s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-facebook-marketing/metadata.yaml | ✅        |
| Connector version semver check                                                   | ✅        |
| Connector version increment check                                                | ❌        |
| QA checks                                                                        | ✅        |
| Code format checks                                                               | ✅        |
| Connector package install                                                        | ✅        |
| Build source-facebook-marketing docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                       | ✅        |
| Integration tests                                                                | ✅        |
| Acceptance tests                                                                 | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/facebook-marketing-spec/1692022847/3a4ce7179ded844658c07e09fa6cd51df97652a3/source-facebook-marketing/1.1.3/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:3a4ce7179ded844658c07e09fa6cd51df97652a3)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-facebook-marketing test
```

**Comment by octavia-squidington-iii** [comment]
##  source-facebook-marketing test report (commit [`5330267f9f`](https://github.com/airbytehq/airbyte/pull/28948/commits/5330267f9fae36223e5b26647514c052112b3e73)) - ❌

⏲️  Total pipeline duration: 76mn01s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-facebook-marketing/metadata.yaml | ✅        |
| Connector version semver check                                                   | ✅        |
| Connector version increment check                                                | ✅        |
| QA checks                                                                        | ✅        |
| Code format checks                                                               | ✅        |
| Connector package install                                                        | ✅        |
| Build source-facebook-marketing docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                       | ✅        |
| Integration tests                                                                | ✅        |
| Acceptance tests                                                                 | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/facebook-marketing-spec/1692026754/5330267f9fae36223e5b26647514c052112b3e73/source-facebook-marketing/1.1.4/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:5330267f9fae36223e5b26647514c052112b3e73)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-facebook-marketing test
```

File: airbyte-integrations/connectors/source-facebook-marketing/integration_tests/spec.json [hunk 1]
```diff
@@ -1,456 +1,466 @@
...
-                "type": "boolean"
+            "fields": {
+              "title": "Fields",
+              "description": "Use the dropdown menu to add the desired fields for your custom insight.",
+              "default": [],
+              "order": 2,
+              "type": "array",
+              "items": {
+                "title": "ValidEnums",
+                "description": "An enumeration.",
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-facebook-marketing/integration_tests/spec.json:99]: @ChristoGrab I fixed the acceptance test issue ( I believe) but there's a number of fields whose titles and descriptions aren't helpful like this one. Basically anything whose description is `An enumeration` could be better explained. Mind taking another look at those fields?

Comment (sherifnada) [reply airbyte-integrations/connectors/source-facebook-marketing/integration_tests/spec.json:99]: Once you fix this, run `python main.py spec | jq . > integration_tests/spec.json` to update the test suite to expect the updated spec

for more info on why we are updating this file see [this page](https://docs.airbyte.com/connector-development/testing-connectors/connector-acceptance-tests-reference/#test-spec) (the whole page is helpful to contextualize acceptance testing)

**Review by sherifnada — CHANGES_REQUESTED** [review]

**Comment by octavia-squidington-iii** [comment]
##  source-facebook-marketing test report (commit [`2b4e615269`](https://github.com/airbytehq/airbyte/pull/28948/commits/2b4e615269cc1ec603255eedcd2d4d5d30383999)) - ✅

⏲️  Total pipeline duration: 61mn02s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-facebook-marketing/metadata.yaml | ✅        |
| Connector version semver check                                                   | ✅        |
| Connector version increment check                                                | ✅        |
| QA checks                                                                        | ✅        |
| Code format checks                                                               | ✅        |
| Connector package install                                                        | ✅        |
| Build source-facebook-marketing docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                                       | ✅        |
| Integration tests                                                                | ✅        |
| Acceptance tests                                                                 | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/facebook-marketing-spec/1692223531/2b4e615269cc1ec603255eedcd2d4d5d30383999/source-facebook-marketing/1.1.4/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:2b4e615269cc1ec603255eedcd2d4d5d30383999)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-facebook-marketing test
```

File: airbyte-integrations/connectors/source-facebook-marketing/integration_tests/spec.json (hunk 1 above)
Comment (ChristoGrab) [reply airbyte-integrations/connectors/source-facebook-marketing/integration_tests/spec.json]: I had to do some digging to figure out where this description was coming from, it turns out it was being generated by the Enum class that comes packaged with Python. Worth noting that these titles/descriptions are never displayed to the user in the Airbyte UI, but I've tried to make them more descriptive for anyone looking through the JSON schema :)

**Review by sherifnada — APPROVED** [review]
oh thanks, I didn't realize these were not visible in the UI

**Comment by airbyte-oss-build-runner** [comment]
##  source-facebook-marketing test report (commit [`3dd0735b62`](https://github.com/airbytehq/airbyte/pull/28948/commits/3dd0735b6233143e579e37098bb71791616872a0)) - ❌

⏲️  Total pipeline duration: 113mn17s 

| Step                                                                      | Result   |
|:--------------------------------------------------------------------------|:---------|
| Connector package install                                                 | ✅        |
| Build source-facebook-marketing docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                                | ✅        |
| Integration tests                                                         | ✅        |
| Acceptance tests                                                          | ❌        |
| Code format checks                                                        | ✅        |
| Validate metadata for source-facebook-marketing                           | ✅        |
| Connector version semver check                                            | ✅        |
| Connector version increment check                                         | ❌        |
| QA checks                                                                 | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/facebook-marketing-spec/1696534690/3dd0735b6233143e579e37098bb71791616872a0/source-facebook-marketing/1.1.14/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:3dd0735b6233143e579e37098bb71791616872a0)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-facebook-marketing test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-facebook-marketing test report (commit [`76616161e0`](https://github.com/airbytehq/airbyte/pull/28948/commits/76616161e08239efb1dfbf3ade280db84928b1e9)) - ❌

⏲️  Total pipeline duration: 110mn38s 

| Step                                                                      | Result   |
|:--------------------------------------------------------------------------|:---------|
| Connector package install                                                 | ✅        |
| Build source-facebook-marketing docker image for platform(s) linux/x86_64 | ✅        |
| Unit tests                                                                | ✅        |
| Integration tests                                                         | ✅        |
| Acceptance tests                                                          | ❌        |
| Code format checks                                                        | ✅        |
| Validate metadata for source-facebook-marketing                           | ✅        |
| Connector version semver check                                            | ✅        |
| Connector version increment check                                         | ✅        |
| QA checks                                                                 | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/facebook-marketing-spec/1696542266/76616161e08239efb1dfbf3ade280db84928b1e9/source-facebook-marketing/1.1.15/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:76616161e08239efb1dfbf3ade280db84928b1e9)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-facebook-marketing test
```

**Pull request closed without merging**